javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/OpenTelemetrySdkAccess.java

```diff
@@ -45,5 +46,32 @@ public static void internalSetForceFlush(ForceFlusher forceFlush) {
     OpenTelemetrySdkAccess.forceFlush = forceFlush;
   }
 
+  /**
+   * Sends copies of trigger and function spans early.
+   */
+  public interface EarlySpans {
+    /** Sends copies of trigger and function spans early. */
+    void sendEarlySpans(Context upstreamContext, Context triggerContext, Context functionContext);
+  }
+
+  private static volatile EarlySpans earlySpans;
+
+  /** Sends copies of trigger and function spans early. */
+  public static void sendEarlySpans(Context upstreamContext, Context triggerContext, Context functionContext) {
+    earlySpans.sendEarlySpans(upstreamContext, triggerContext, functionContext);
+  }
+
+  /**
+   * Sets the earlySpans. This is called
+   * from the agent class loader to execute the SDK's mechanism. Instrumentation must not call this.
+   */
+  public static void internalSetEarlySpans(EarlySpans earlySpans) {
+    if (OpenTelemetrySdkAccess.earlySpans != null) {
```

Comment:
To make it clearer it could be other way round: if null then assign (as strange as it seems but I guess it has to be).